Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -1,20 +1,38 @@
 import UIKit
+import CoreLocation
 
-// MARK: CameraAnimator Class
-public class CameraAnimator: NSObject {
+@objc public protocol CameraAnimatorProtocol: AnyObject {
+
+    /// Stops the animation in its tracks and calls any provided completion
+    func stopAnimation()
 
-    // MARK: Stored Properties
+    /// The current state of the animation
+    var state: UIViewAnimatingState { get }
+
+    /// TODO: Move this to a non-public interface
+    func update()
+}
+
+// MARK: CameraAnimator Class
+public class CameraAnimator: NSObject, CameraAnimatorProtocol {
 
     /// Instance of the property animator that will run animations.
-    private var propertyAnimator: UIViewPropertyAnimator
+    internal var propertyAnimator: UIViewPropertyAnimator
 
     /// Delegate that conforms to `CameraAnimatorDelegate`.
-    private weak var delegate: CameraAnimatorDelegate?
+    internal weak var delegate: CameraAnimatorDelegate?
 
     /// The ID of the owner of this `CameraAnimator`.
-    internal var owner: AnimationOwner
+    public internal(set) var owner: AnimationOwner
+
+    /// The `CameraView` owned by this animator
+    internal var cameraView: CameraView
```

Comment:
Could this be private?

Reply:
it could be `internal private(set)`.. we do use a mock of it in some tests.

Reply:
nvm we can make it private.